Sparkle/SUPlainInstaller.m

```diff
@@ -7,13 +7,165 @@
 //
 
 #import "SUPlainInstaller.h"
-#import "SUPlainInstallerInternals.h"
+#import "SUFileManager.h"
 #import "SUCodeSigningVerifier.h"
 #import "SUConstants.h"
 #import "SUHost.h"
+#import "SULog.h"
 
 @implementation SUPlainInstaller
 
+// Returns the bundle version from the specified host that is appropriate to use as a filename, or nil if we're unable to retrieve one
++ (NSString *)bundleVersionAppropriateForFilenameFromHost:(SUHost *)host
+{
+    NSString *bundleVersion = [host objectForInfoDictionaryKey:(NSString *)kCFBundleVersionKey];
+    NSString *trimmedVersion = @"";
+    
+    if (bundleVersion != nil) {
+        NSMutableCharacterSet *validCharacters = [NSMutableCharacterSet alphanumericCharacterSet];
+        [validCharacters formUnionWithCharacterSet:[NSCharacterSet characterSetWithCharactersInString:@".-()"]];
+        
+        trimmedVersion = [bundleVersion stringByTrimmingCharactersInSet:[validCharacters invertedSet]];
+    }
+    
+    return trimmedVersion.length > 0 ? trimmedVersion : nil;
+}
+
++ (BOOL)performInstallationToURL:(NSURL *)installationURL fromUpdateAtURL:(NSURL *)newURL withHost:(SUHost *)host error:(NSError * __autoreleasing *)error
+{
+    SUFileManager *fileManager = [[SUFileManager alloc] init];
+    
+    // Create a temporary directory for our new app that resides on our destination's volume
+    NSURL *tempNewDirectoryURL = [fileManager makeTemporaryDirectoryWithPreferredName:[installationURL.lastPathComponent.stringByDeletingPathExtension stringByAppendingString:@" (Incomplete Update)"] appropriateForDirectoryURL:installationURL.URLByDeletingLastPathComponent error:error];
```

Comment:
Since we're installing a single item, having a temp directory seems unnecessary.

Could you just move `/tmp/foo.app` to `destination/foo (incomplete vX).app`? or maybe to `.foo (incomplete vX).app` (with a `.` prefix to prevent temp file from popping up in Finder)